torch/utils/data/dataset.py

```diff
@@ -219,9 +215,15 @@ def __add__(self, other: Dataset[T_co]):
     # No `def __len__(self)` default? Subclasses raise `TypeError` when needed.
     # See NOTE [ Lack of Default `__len__` in Python Abstract Base Classes ]
 
+
+class IterDataPipe(IterableDataset[T_co], metaclass=_DataPipeMeta):
+    functions: Dict[str, Callable] = {}
+    reduce_ex_hook : Optional[Callable] = None
+    getstate_hook: Optional[Callable] = None
```

Comment:
Yeah, this is awesome! Just a note that we may need to change here https://github.com/pytorch/pytorch/blob/29d759948ee7cd2275ea60ebd62c311408e3798b/torch/utils/data/__init__.py#L19

Reply:
NVM, I didn't notice you have already changed it.